src/entity/Unit.ts

```diff
@@ -1452,25 +1452,23 @@ export function findLOSLocation(unit: IUnit, target: Vec2, underworld: Underworl
 
 
 // handles drawing attack range, bloat radius, and similar graphics each frame while a unit is selected
-export async function drawSelectedGraphics(unit: IUnit, prediction: boolean = false, underworld: Underworld): Promise<void> {
+export function drawSelectedGraphics(unit: IUnit, prediction: boolean = false, underworld: Underworld) {
 
-  if (!prediction) {
-    for (let drawEvent of unit.onDrawSelectedEvents) {
-      if (drawEvent) {
-        const fn = Events.onDrawSelectedSource[drawEvent];
-        if (fn) {
-          await fn(unit, prediction, underworld);
-        } else {
-          console.error('No function associated with onDrawSelected event', drawEvent);
-        }
+  if (globalThis.headless || prediction || !globalThis.selectedUnitGraphics) return;
+
+  for (let drawEvent of unit.onDrawSelectedEvents) {
+    if (drawEvent) {
+      const fn = Events.onDrawSelectedSource[drawEvent];
+      if (fn) {
+        fn(unit, prediction, underworld);
+      } else {
+        console.error('No function associated with onDrawSelected event', drawEvent);
       }
     }
   }
 
-  if (!globalThis.selectedUnitGraphics) {
-    console.error("selectedUnitGraphics is undefined")
-    return;
-  }
+  // TODO - Ideally the logic below would be defined in each Unit's ts file individually
+  // Instead of using if/else and unit subtypes
```

Comment:
Why? If the logic is always the same for each given subtype wouldn't that just result in a lot of repetition for units that share the same subtype?

Reply:
This would be for the same reason we put .action or .getUnitAttackTargets on each unit instead of in a large if/else. 

There are already many different ways a unit can draw information, and I imagine there will only be more as new units are added, especially when considering different kinds of doodads, aoe enemies, spellcasters, and so on. Many of the current drawings could be separated and improved too, such as the glops and priests. It would be nice to show more than just the attack range circle for most enemies, and I think that defining it in each enemy's script would be a cleaner way of doing this than to have the unit specific logic be in the broader unit class (and onDrawSelected is currently the only place this happens, afaik)

It also separates drawing behavior from unit.subtypes, making the subtypes more of an identification/targeting group thing, which I think is another step in the right direction.

Reply:
Yeah that sounds good to me